justfile

```
@@ -13,12 +13,41 @@ default_docker_tag := 'local'
 default_repo_name := 'ghcr.io/astriaorg'
 
 # Builds docker image for the crate. Defaults to 'local' tag.
-docker-build crate tag=default_docker_tag repo_name=default_repo_name:
-  docker buildx build --load --build-arg TARGETBINARY={{crate}} -f containerfiles/Dockerfile -t {{repo_name}}/$(echo {{crate}} | sed 's/^[^-]*-//'):{{tag}} .
-
-docker-build-and-load crate tag=default_docker_tag repo_name=default_repo_name:
-  @just docker-build {{crate}} {{tag}} {{repo_name}}
-  @just load-image $(echo {{crate}} | sed 's/^[^-]*-//') {{tag}} {{repo_name}}
+# NOTE: `_crate_short_name` is invoked as dependency of this command so that failure to pass a valid
+# crate will produce a maningful error message.
+docker-build crate tag=default_docker_tag repo_name=default_repo_name: (_crate_short_name crate "quiet")
+  #!/usr/bin/env sh
+  set -eu
+  short_name=`just _crate_short_name {{crate}}`
+  set -x
+  docker buildx build --load --build-arg TARGETBINARY={{crate}} -f containerfiles/Dockerfile -t {{repo_name}}/$short_name:{{tag}} .
+
+# Builds and loads docker image for the crate. Defaults to 'local' tag.
+# NOTE: `_crate_short_name` is invoked as dependency of this command so that failure to pass a valid
+# crate will produce a maningful error message.
+docker-build-and-load crate tag=default_docker_tag repo_name=default_repo_name: (_crate_short_name crate "quiet")
+  #!/usr/bin/env sh
+  set -eu
+  short_name=`just _crate_short_name {{crate}}`
+  set -x
+  just docker-build {{crate}} {{tag}} {{repo_name}}
+  just load-image $short_name {{tag}} {{repo_name}}
+
+# Maps a crate name to the shortened name used in the docker tag.
+# If `quiet` is an empty string the shortened name will be echoed. If `quiet` is a non-empty string,
+# the only output will be in the case of an error, where the input `crate` is not a valid one.
+_crate_short_name crate quiet="":
+  #!/usr/bin/env sh
+  set -eu
+  case {{crate}} in
+    astria-bridge-withdrawer) short_name=bridge-withdrawer ;;
+    astria-composer) short_name=composer ;;
+    astria-conductor) short_name=conductor ;;
+    astria-sequencer) short_name=sequencer ;;
+    astria-sequencer-relayer) short_name=sequencer-relayer ;;
+    *) echo "{{crate}} is not a supported crate" && exit 2
```

I think we should call this binary because at the level of the dockerfiles users should not have to care about Rust idioms.
```suggestion
    *) echo "{{crate}} is not a supported binary" && exit 2
```

---

I'll make the change suggested here, but I don't want to change all usages of the term "crate" in this file (some of which predate this PR).  I think "crate" is ubiquitous in Rust, to the point that if a reader is seeing `cargo` being used in this file, they'll very likely understand the term "crate" in this context.

---

Yeah, most of the usage I consider fine - just the outside facing messages I think should not be about rust (focusing on the output, the service/image, instead of the input).

---

Done in [c88db3b](https://github.com/astriaorg/astria/pull/1804/commits/c88db3b0d0a5d043baaf0816dbceea307555dec8).